src/download.c

```diff
@@ -306,18 +310,51 @@ static int perform_curl_io_and_complete(struct swupd_curl_parallel_handle *h, in
 	return 0;
 }
 
-/* Fixed window added to hysteresis upper bound before it is enforced. */
-#define BUFFER 5
+static CURLMcode process_curl_multi_wait(CURLM *mcurl, int *repeats)
+{
+	CURLMcode ret;
+	int numfds = 0;
+
+	ret = curl_multi_wait(mcurl, NULL, 0, CURL_MULTI_TIMEOUT, &numfds);
+
+	// If numfds is zero curl_multi_wait reached a timeout or there isn't
+	// any fd to process. The first ocurrence we assume it's a timeout.
+	// After that we assume we need to force a sleep before calling
+	// curl_multi_wait() again.
+	if (ret != CURLM_OK && !numfds) {
+		if (*repeats == 1) {
+			usleep(100000); // sleep 100ms
```

Comment:
you should #define this like you did with CURL_MULTI_TIMEOUT. It could be NOFD_WAIT or something